rhoas/kafkas/resource_kafka.go

```diff
@@ -122,6 +123,9 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 		Refresh: func() (interface{}, string, error) {
 			data, resp, err1 := api.KafkaMgmt().GetKafkaById(ctx, d.Id()).Execute()
 			if err1 != nil {
+				if err1.Error() == "404 Not Found" {
+					return data, "404", nil
+				}
 				if apiErr := utils.GetAPIError(resp, err); apiErr != nil {
```

Comment:
Need to pass err1 here to GetAPIError not err

Reply:
true, updated